### ISSUE TITLE
image.quantize(2) does not work same on all images 

### ISSUE THREAD

**Issue by Samuel-Bachorik** [python-pillow/Pillow#5690]
Hi I have issue with quantize() method. I am using this function in images loader class function. This mask post processing function (I mentioned down) is working with masks for neural network. I am loading 1000 similar images and processing them. This function quantize() does not work same on all of them. 
Here is link for my dataset. Images and mask are all same format-
https://drive.google.com/drive/u/1/folders/1795opF54wK76r5snXs68OtGl2cR7g-3C

Example:
Some masks of road looks like this -
And this is how it should looks like
![image](https://user-images.githubusercontent.com/61843287/131080908-9c8c8b55-bf32-4ab8-bfc7-f9aa212e1d2a.png)

But some of images lost second class after quantize() method affects them 
Like this - 
![image](https://user-images.githubusercontent.com/61843287/131080789-23520c9d-3e27-419e-b3a8-78165f2933b8.png)

Here is images loader and at line 290 I am using quantize() and there some of my images lose the second class

I am using image.show() before and after quantize() some of images are good and some bad as i showed on example
https://github.com/Samuel-Bachorik/PyTorch-AI-road-detection-classification/blob/d3cf9fd5224b1a1032479d5dea7a6c3a0ba37d78/Model_loader.py#L290-L299

**Comment by radarhere** [comment]
So when I run this code
```python
from PIL import Image

for file, out in (
	("20210303_103546_watershed_mask.png", "46.png"),
	("20210303_103547_watershed_mask.png", "47.png")
):
	print(file)
	image = Image.open(file)
	image = image.quantize(2).convert("RGB")
	px = image.load()
	colors = []
	for x in range(image.width):
		for y in range(image.height):
			if px[x, y] not in colors:
				colors.append(px[x, y])
	print("Number of colors:", len(colors))
	for x in range(image.width):
		for y in range(image.height):
			px[x, y] = (0, 0, 255) if px[x, y] == (8, 8, 8) else (0, 0, 0)
	image.save(out)
```
and receive these images
![46](https://user-images.githubusercontent.com/3112309/131242894-54777d8c-5d7d-48d5-bfed-3e468924d830.png)
![47](https://user-images.githubusercontent.com/3112309/131242895-25da767d-110e-48d2-856f-164485845e0d.png)

That's the correct output, yes? But you're saying that you are aware that this works correctly outside of your script?

**Comment by Samuel-Bachorik** [comment]
@radarhere Yes your output for both images is right. 

> But you're saying that you are aware that this works correctly outside of your script?

Yes exactly, when try to use quantize() for these images in another file like you, it is working fine. Only in this images loader class it is not working properly. Some of images from dataset it process right and some not.

**Comment by Samuel-Bachorik** [comment]
@radarhere Yes your output for both images is right. 

> But you're saying that you are aware that this works correctly outside of your script?

Yes exactly, when try to use quantize() for these images in another file like you, it is working fine. Only in this images loader class it is not working properly. Some of images from dataset it process right and some not.

**Comment by Samuel-Bachorik** [comment]
@radarhere  Hi, I was working on this issue and I found where problem is. Problem is in dataset, if you look carefully on _watershed_masks you will see these masks have frame. I figured it out and created script that delete this frame for me before it goes to my images loader class and it worked. 

So it is solved now, thank you for your time a lot! 

Samuel
